teuthology/task/ceph_ansible.py

```diff
@@ -61,25 +74,69 @@ class CephAnsible(ansible.Ansible):
     )
 
     def __init__(self, ctx, config):
+        super(CephAnsible, self).__init__(ctx, config)
         config = config or dict()
         if 'playbook' not in config:
-            config['playbook'] = self._default_playbook
+            self.playbook = self._default_playbook
+        else:
+            self.playbook = self.config['playbook']
         if 'repo' not in config:
-            config['repo'] = os.path.join(teuth_config.ceph_git_base_url,
-                                          'ceph-ansible.git')
-        super(CephAnsible, self).__init__(ctx, config)
+            self.config['repo'] = os.path.join(teuth_config.ceph_git_base_url,
+                                               'ceph-ansible.git')
+
+    def setup(self):
+        super(CephAnsible, self).setup()
+        # generate hosts file based on test config
+        self.generate_hosts_file()
+        # use default or user provided playbook file
+        pb_buffer = StringIO()
+        pb_buffer.write('---\n')
+        yaml.safe_dump(self.playbook, pb_buffer)
+        pb_buffer.seek(0)
+        playbook_file = NamedTemporaryFile(
+            prefix="ceph_ansible_playbook_",
+            dir='/tmp/',
+            delete=False,
+        )
+        playbook_file.write(pb_buffer.read())
+        playbook_file.flush()
+        self.playbook_file = playbook_file
 
-    def get_inventory(self):
+    def execute_playbook(self, _logfile=None):
         """
-        Stub this method so we always generate the hosts file
+        Execute ansible-playbook
+
+        :param _logfile: Use this file-like object instead of a LoggerFile for
+                         testing
         """
-        pass
+
+        # everything from vars in config go into extra-vars
+        # TODO add option to directly copy them to group_vars/all
+        extra_vars = dict()
+        extra_vars.update(self.config.get('vars', dict()))
+        args = [
+            'ansible-playbook', '-v',
+            "--extra-vars", "'%s'" % json.dumps(extra_vars),
```

Comment:
Do we plan to allow for ``group_vars``?  Using ``--extra-vars`` will clobber any configuration settings set in ``group_vars`` or ``host_vars``.